### ISSUE TITLE
Problem reconnecting to jupyter notebook

### ISSUE THREAD

**Issue by matssk** [microsoft/vscode-jupyter#16323]
### Applies To

- [x] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

If the jupyter notebook server restarts, the plugin refuses to reconnect until I restart Visual Studio Code. It says "Unable to connect to the remote server. Ensure the server is running and reachable." This error message will persist until Visual Studio Code is restarted, even though the jupyter server is already back up. 

### VS Code Version

1.96

### Jupyter Extension Version

 2024.11.0

### Jupyter logs

```shell

```

### Running Jupyter locally or remotely?

Remote

**Comment by DonJayamanne** [comment]
Please could you enable logging as follows:
* Open settings & go into `Jupyter->Logging`
* Change the value to `trace`
* Reload VS Code,
* Attempt to repro the issue & then copy the output from the `Jupyter` output panel (use the command `Jupyter: Show Output` to get to the logs).

**Comment by matssk** [comment]
I won't be able to do that for a few weeks, but I was looking a bit at server logs and it looks like maybe what's happening is it tries logging in again with the same cookies/headers, but these are not valid any more since the server restarted, so it should retry with the token and it seems maybe it's not doing that unless vscode is restarted.

**Comment by DonJayamanne** [comment]
Do you have a notebook connected to a kennel?
If you do, and then restart the Jupiter server then the kernel that was connected to the notebook is now lost and reconnecting doesn't work.

Hence you need to reselect a kennel spec to get around this.

Is this what you are running into.

**Comment by matssk** [comment]
Yes. After restarting the jupyter server I click "Select kernel", then "Existing Jupyter server" and I choose the server there. When I try to reselect the kernel and press the reload icon, it does not connect. After I restart vscode I can try reconnecting to the kernel and it works, but I have to enter the token again.

**Comment by DonJayamanne** [comment]
> and it works, but I have to enter the token again.

Oh so you enter a while new token, is that correct?
And the request is to be provided to enter the new token instead of having to restart vscode.

**Comment by matssk** [comment]
I don't get a request for a new token unless I restart vscode, it just says it can't connect.
